src/twisted/web/test/test_flatten.py

```diff
@@ -433,6 +433,28 @@ def test_serializeDeferredSlots(self):
         return self.assertFlattensTo(t, b'<p><em>four&gt;</em></p>')
 
 
+    def test_slotDataPropagation(self):
+        """
+        slot data propagation is only made to child nodes.
+        Before #9383 fix, slot datas were not correctly clean during tree
+        traversal and some elements that are not directly child nodes can
+        reach forbidden slot data.
```

Comment:
What is a forbidden slot data?

I have never worked with this API...so I am not sure about the terminology.

But to me, it looks like this test is about.

```
Slot data is applied only to the targeted tag.
```